chain-db-watcher/src/graphql_helpers.ts

```diff
@@ -0,0 +1,127 @@
+import dotenv from 'dotenv';
+import { GraphQLClient } from 'graphql-request';
+
+import { addPostAndProposalMutation, getProposalQuery, loginMutation } from './queries';
+
+dotenv.config();
+
+const graphqlServerUrl = process.env.REACT_APP_HASURA_GRAPHQL_URL;
+
+/**
+ * Tell if there is already a proposal in the DB matching the
+ * onchain proposal id passed as argument
+ *
+ * @param onchain_proposal_id the proposal id that is on chain (not the Prisma db id)
+ */
+export const proposalAlreadyExists = async (onchain_proposal_id: number): Promise<boolean> => {
+
+	if (!graphqlServerUrl) throw new Error ('Environment variable for the REACT_APP_HASURA_GRAPHQL_URL not set');
+
+	try {
+		const client = new GraphQLClient(graphqlServerUrl, {
+			headers: {}
+		});
+
+		return client.request(getProposalQuery, { onchain_proposal_id })
+			.then(data => !!data?.onchain_proposals?.length)
```

Comment:
!!(data?.onchain_proposals?.length) to be more specific

Reply:
I wasn't convinced about this one, and eslint (with no extra parens) is actually saying that those parenthesis aren't necessary.

Reply:
ok